client/src/App.js

```diff
@@ -22,11 +22,10 @@ export function App() {
 	}
 	, []);
 
-
-	return students ?(
+	return students ? (
 		<Router>
 			<Route exact path="/students" component={() => <StudentsGrid students={students} />} />
-			<Route exact path="/students/:name" component={() => <StudentProfile students={students} />} />
```

Comment:
I don't think we need to pass students as props to StudentProfile component, as we are fetching again in StudentProfile for the specific student. 
If we pass the props e.g. not to fetch again then we could find the specific student with a function.